vm/inc/ubpf.h

```diff
@@ -260,4 +263,31 @@ ubpf_get_registers(const struct ubpf_vm* vm);
 int
 ubpf_set_pointer_secret(struct ubpf_vm* vm, uint64_t secret);
 
+typedef uint64_t (*ubpf_data_relocation)(
+    void* user_context, const uint8_t* data, uint64_t data_size, const char* symbol_name);
+
```

Comment:
My only request -- could we reformulate the parameters of the callback slightly?

- `user_context` (same signature): No change
- `data` (same signature): A pointer to the start of the section targeted
- `data_size` (`uint64_t`): The size of the section targeted
- `symbol_offset` (uint64_t): The offset into the section targeted where the symbol exists
- `symbol_size` (uint64_t): The size of the symbol
 
That way, it would be easy for the callback to recognize when it is being asked to handle a relocation into a section it has never seen (will need to copy that *entire* section to handle accesses at runtime) and one that it has already seen (and, therefore, has already copied to handle accesses at runtime).

Reply:
Just wondering if the omission of the symbol size parameter was intentional. If so, that's great -- you are the expert. Otherwise, I just wanted to flag it.